graviti/dataframe/row/series.py

```diff
@@ -54,54 +51,66 @@ def __init__(
             # TODO: missing index processing
             pass
 
-        self._data = []
-        self._indices_data = {}
+        self._indices_data, self._indices = {}, []
         for key, value in data.items():
             if isinstance(value, dict):
                 value = Series(value, name=name)
-            self._data.append(value)
             self._indices_data[key] = value
-        self._indices = list(data.keys())
+            self._indices.append(key)
         self.name = name
 
     # @overload
     # def __getitem__(self, key: slice) -> "Series":
     #    ...
 
     @overload
-    def __getitem__(self, key: Tuple[_T]) -> "Series[_T]":
+    def __getitem__(self, key: Tuple[str]) -> "Series":
         ...
 
     @overload
-    def __getitem__(self, key: _T) -> Any:
+    def __getitem__(self, key: str) -> Any:
         ...
 
     @overload
-    def __getitem__(self, key: Iterable[_T]) -> "Series[_T]":
+    def __getitem__(self, key: Iterable[str]) -> "Series":
         ...
 
-    def __getitem__(self, key: Union[_T, Iterable[_T]]) -> Any:
-        if isinstance(key, (str, int)):
+    def __getitem__(self, key: Union[str, Iterable[str]]) -> Any:
+        if isinstance(key, str):
             return self._indices_data[key]
 
         new_data = {name: self._indices_data[name] for name in key}
         return Series(new_data, name=self.name)
 
+    def __len__(self) -> int:
+        return self._indices.__len__()
+
     # @overload
     # def _getitem_by_location(self, key: slice) -> "Series":
     #    ...
 
     @overload
-    def _getitem_by_location(self, key: int) -> Union["Series[_T]", Any]:
+    def _getitem_by_location(self, key: int) -> Union["Series", Any]:
         ...
 
     @overload
-    def _getitem_by_location(self, key: Iterable[int]) -> "Series[_T]":
+    def _getitem_by_location(self, key: Iterable[int]) -> "Series":
         ...
 
     def _getitem_by_location(self, key: Union[int, Iterable[int]]) -> Any:
         if isinstance(key, int):
-            return self._data[key]
-
-        new_data = {self._indices[index]: self._data[index] for index in key}
-        return Series(new_data, name=self.name)
+            return self._indices_data[self._indices[key]]
+
+        indices_data = {
+            self._indices[index]: self._indices_data[self._indices[index]] for index in key
+        }
+        return self._constructer(indices_data, self.name)
+
+    @staticmethod
+    def _constructer(indices_data: Dict[str, Any], new_name: Union[str, int, None]) -> "Series":
```

Comment:
This looks like a `classmethod`, and should `DataFrame._constructer` be a classmethod?

Reply:
and normally we do not name a method in class in `-er` style